src/Interpreters/Streaming/HashJoin.cpp

```diff
@@ -904,8 +931,9 @@ void HashJoin::init()
     /// In case when emitChangeLog()
     if (streaming_strictness == Strictness::Range
         && (left_data.join_stream_desc->data_stream_semantic != DataStreamSemantic::Append
-            || right_data.join_stream_desc->data_stream_semantic != DataStreamSemantic::Append))
-        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Stream range join only support append-only streams");
+            || right_data.join_stream_desc->data_stream_semantic != DataStreamSemantic::Append
+            || streaming_kind != Kind::Inner))
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Only support inner range join append-only streams");
```

Comment:
`Only inner range join is supported and the left and right stream must be append-only streams in range join`